src/Views/ProgressView.vala

```diff
@@ -127,12 +127,18 @@ public class ProgressView : AbstractInstallerView {
 
         unowned Configuration current_config = Configuration.get_default ();
 
-        //TODO: Use the following
         debug ("language: %s\n", current_config.lang);
-        if (current_config.country != null) {
+        string lang = current_config.lang;
+        if (current_config.country != null && current_config.country != "") {
             debug ("country: %s\n", current_config.country);
-        } else {
-            config.lang = current_config.lang + "_" + current_config.lang.ascii_up () + ".UTF-8";
+            lang += "_" + current_config.country;
+        }
+
+        string? locale;
+        if (yield LocaleHelper.language2locale (lang, out locale)) {
+            if (locale != null) {
+                config.lang = locale;
+            }
```

Comment:
Do we have some debugging/feedback available in case the language isn't found?

Reply:
I guess this just silently falls back to en_US at the moment. I'm just trying to improve on the existing situation that selects invalid locales. I can add some debug messaging, but I don't think it gets logged anywhere during a real installation?